### DISCUSSION TITLE
Update of 2.0.0 Database to 2.0.1

### DISCUSSION THREAD

**Discussion by dbltoe** [zencart/zencart#6454] [General]
I know I'm not the only one dealing with folks who loaded everything from alpha to RC1 on 2.0.0 and who do not have a current database that is complete.

Rather than having to pick through the file in zc_install for the right changes/creations, why not have 2.0.1 to make sure the database has all the necessary parts?

IMHO, this could solve a big headache for many sites.

**Comment by scottcwilson** [comment]
We're going to do a 2.0.1 build shortly to capture the bug fixes that were made subsequent to the release of 2.0.0. 
Does this answer your question? Are you asking for test sites running 2.0.0 with demo data?  If so they could just re-run zc_install and create a new database.

**Reply by dbltoe** [reply]
I am seeing many sites that have bits and pieces of the database missing that would be standard in a clean 2.0.0.  It appears to be due to folks using alpha, beta, or RC versions.  Next, they start seeing missing items when updating mods to 2.0 versions.
If you look at the blame for the 2.0.0 db upgrade file, you'll find changes almost monthly from 10 months ago to your update last month.  Any person going live during those 10 months could very well wind up with a database that is incomplete and might not play well with newer 2.0.0-capable mods.

**Reply by scottcwilson** [reply]
Anyone who goes live using unreleased code bears the entire burden of keeping things in sync themselves.  This is a strategy that is highly discouraged for anyone other than experts (and daredevils!)

**Reply by dbltoe** [reply]
And, unfortunately, it is a strategy we have little or no control over.  I was just wondering if we could head the problem off without much effort.

**Reply by torvista** [reply]
> Anyone who goes live using unreleased code bears the entire burden of keeping things in sync themselves

Absolutely. Agreed.

Although it's those guinea pigs who are to thank for each release having been tested in the wild and finding anomalies with environments, so should not be completely scorned, but still, on their own head be it.
I'm one, and take that risk, primarily to be able to PR directly on dev code.

**Reply by scottcwilson** [reply]
To be clear: this is an EXPERTS ONLY approach, and I am grateful that there are experts who are willing to bear the risk and exercise this code so as to find bugs early.  I have amended my comments to clarify this. 

Non-experts beware - here be monsters.

**Comment by drbyte** [comment]
It's also important to note that we normally do zero database changes in "bugfix" releases.

ie: 
2.0 - db changes allowed
2.0.1 - no db changes
2.0.2 - no db changes
2.1.0 - db changes allowed
2.1.1 - no db changes
2.2 - db changes allowed
2.2.1 - no db changes

**Comment by drbyte** [comment]
> Rather than having to pick through the file in zc_install for the right changes/creations, why not have 2.0.1 to make sure the database has all the necessary parts?

Probably best to just run the `mysql_upgrade_zencart_200.sql` file via Sql-Patch to catch up everything that was done in various stages of 2.0.0-alpha-to-final-release.

**Comment by dbltoe** [comment]
Understood.  I'm just seeing more of this with this release.  It's probably due to the increased awareness of GitHub. 

I will give the Sql-Patch a shot in the next case.  Navicat is having problems running the upgrade against the database.  It's stopping if just one of the table changes has already been done.

**Reply by drbyte** [reply]
Right. With phpMyAdmin or Navicat you'll have to run each statement one-by-one and just deal with reported duplicates.
In Navicat you can highlight a statement and right-click and run just the highlighted code.

SqlPatch soft-ignores errors if they're duplicate-inserts etc.

**Reply by dbltoe** [reply]
Learn something new everyday.  Gotta try that right-click trick.
